sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/CostBasedJoinReorder.scala

```diff
@@ -202,14 +201,15 @@ object JoinReorderDP extends PredicateHelper {
   }
 
   /**
-   * Builds a new JoinPlan if the two given sides can be joined with some conditions.
+   * Builds a new JoinPlan if there exists at least one join condition involving references from
+   * both left and right.
    * @param oneJoinPlan One side JoinPlan for building a new JoinPlan.
    * @param otherJoinPlan The other side JoinPlan for building a new join node.
    * @param conf SQLConf for statistics computation.
    * @param conditions The overall set of join conditions.
    * @param topOutput The output attributes of the final plan.
-   * @return Return a new JoinPlan if the two sides can be joined with some conditions. Otherwise,
-   *         return None.
+   * @return Builds and returns a new JoinPlan if there exists at least one join condition
+   *         involving references from both left and right. Otherwise, returns None.
```

Comment:
Now, we can simplify it to `Builds and returns a new JoinPlan if both conditions hold. Otherwise, returns None.`